src/main/kotlin/org/radarbase/gateway/Config.kt

```diff
@@ -145,6 +150,39 @@ data class KafkaConfig(
         serialization = serializationDefaults + serialization
     )
 
+    /**
+     * Extract KafkaConfig Admin and Producer entries from environmental variables.
+     * Properties for Producer and Admin are prefixed with "$prefix_PRODUCER_" and "$prefix_ADMIN_" respectively.
+     * Property names are deduced by removing prefixes and replacing underscores with dots.
+     * For instance, the environmental variable "KAFKA_PRODUCER_BOOsTSTRAP_SERVERS" will be
+     * converted to "bootstrap.servers" in the producer configuration. The environmental variable
+     * "KAFKA_ADMIN_DEFAULT_API_TIMEOUT_MS" will be converted to "default.api.timeout.ms" in the admin configuration.
+     *
+     * @param prefix the prefix of the environmental variables (usually "KAFKA_")
+     */
+    fun withEnv(prefix: String): KafkaConfig {
+        val admin = admin.toMutableMap()
+        val producer = producer.toMutableMap()
+        System.getenv().entries.stream()
+            .filter { it.key.startsWith(prefix) }
```

Comment:
```suggestion
            .filter { it.key.startsWith(prefix) && it.value.isNotEmpty() }}
```
Otherwise it looks like env variables always override the default values from the config file, even when an env variable has an empty value.

Reply:
I accepted your proposal.